src/Nonlinear/ReverseAD/mathoptinterface_api.jl

```diff
@@ -216,13 +216,16 @@ function MOI.eval_constraint_jacobian(d::NLPEvaluator, J, x)
 end
 
 function MOI.eval_constraint_jacobian_product(d::NLPEvaluator, y, x, w)
+    _reverse_mode(d, x)
     fill!(y, 0.0)
-    J_struct = MOI.jacobian_structure(d)
-    nnz = length(J_struct)
-    J = zeros(nnz)
-    MOI.eval_constraint_jacobian(d, J, x)
```

Comment:
@blegat the main benefit is that we remove the allocations of `J_struct` and `J` from every call.

Reply:
Yes but in https://github.com/jump-dev/JuMP.jl/pull/2239, @amontoison was mentioning a bit slowdown

Reply:
It's not the same implementation. This is purely a refactor that avoids some allocations. The run-time (excluding allocations) is the same.

Reply:
Ok, I don't know the internals well enough to review but I don't have any objection then :)